tests/component/test_write_properties.py

```diff
@@ -0,0 +1,126 @@
+from nidaqmx.task import Task
+
+
+@pytest.fixture(scope="function")
+def write_task(any_x_series_device):
+    """Gets AO voltage task."""
+    with nidaqmx.Task() as task:
+        task.ao_channels.add_ao_voltage_chan(any_x_series_device.ao_physical_chans[0].name)
+        yield task
+
+
+def test__write__get_int32_property__returns_value(write_task: Task):
+    """Test to validate getter for int32 property."""
+    assert write_task.out_stream.offset == 0
+
+
+def test__write__set_int32_property__returns_assigned_value(write_task: Task):
+    """Test to validate setter for int32 property."""
+    write_task.out_stream.offset = 1
+
+    assert write_task.out_stream.offset == 1
+
+
+def test__write__reset_int32_property__returns_default_value(write_task: Task):
+    """Test to validate reset for int32 property."""
+    write_task.out_stream.offset = 2
+
+    del write_task.out_stream.offset
+
+    assert write_task.out_stream.offset == 0
+
+
+def test__write__get_enum_property__returns_value(write_task: Task):
+    """Test to validate getter for enum property."""
+    assert write_task.out_stream.relative_to == WriteRelativeTo.CURRENT_WRITE_POSITION
+
+
+def test__write__set_enum_property__returns_assigned_value(write_task: Task):
+    """Test to validate setter for enum property."""
+    write_task.out_stream.relative_to = WriteRelativeTo.FIRST_SAMPLE
+
+    assert write_task.out_stream.relative_to == WriteRelativeTo.FIRST_SAMPLE
+
+
+def test__write__reset_enum_property__returns_default_value(write_task: Task):
+    """Test to validate reset for enum property."""
+    write_task.out_stream.relative_to = WriteRelativeTo.FIRST_SAMPLE
+
+    del write_task.out_stream.relative_to
+
+    assert write_task.out_stream.relative_to == WriteRelativeTo.CURRENT_WRITE_POSITION
+
+
+def test__write__get_float_property__returns_value(write_task: Task):
+    """Test to validate getter for float property."""
+    assert write_task.out_stream.sleep_time == 0.001
+
+
+def test__write__set_float_property__returns_assigned_value(write_task: Task):
+    """Test to validate setter for float property."""
+    write_task.out_stream.sleep_time = 1
+
+    assert write_task.out_stream.sleep_time == 1
+
+
+def test__write__reset_float_property__returns_default_value(write_task: Task):
+    """Test to validate reset for float property."""
+    write_task.out_stream.sleep_time = 3
+
+    del write_task.out_stream.sleep_time
+
+    assert write_task.out_stream.sleep_time == 0.001
+
+
+def test__write__get_uint32_property__returns_value(write_task: Task):
+    """Test to validate getter for uInt32 property."""
+    assert write_task.out_stream.num_chans == 1
+
+
+def test__write__get_uint64_property__returns_value(write_task: Task):
+    """Test to validate getter for uInt64 property."""
+    write_task.start()
+        
+    assert write_task.out_stream.curr_write_pos == 0
+
+
+@pytest.mark.parametrize("device_by_name", ["nidaqmxUSBTester"], indirect=True)
```

Comment:
When I see the device name "nidaqmxUSBTester", I have no idea what kind of device it is.

For a USB-9265, a good name would be "aoCurrentTester". I think the channel type is more important than the fact it's a DAQmx device (they all are) or that it's connected via USB. For a PXIe-4322, a good name would be "aoTester".

Name these test cases "test__ao_current_task__..." because they require a different task than the rest.

Reply:
Thank you. Replaced `USB-9265` with `PXIe-4322`, all related tests passed. Also, updated the names as well.